src/pages/chat/Chat.tsx

```diff
@@ -92,6 +94,14 @@ const Chat = () => {
     }
   }, [selectedChat, messages.length])
 
+  useEffect(() => {
+    if (currentChatId) {
+      listOfChats.forEach((chat: ChatResponse) => {
+        if (chat._id === currentChatId) return setSelectedChat(chat)
+      })
```

Comment:
Here you use 'return' it is useless. 
And use maybe for loop instead of forEach, because when you find a certain chat and set chat in setSelectedChat(chat) you can break iteration.
forEach doesn't work break.

For example: It will be the first chat but you have fifty chats and when you use forEach it will run through the entire array.

Reply:
> Here you use 'return' it is useless. And use maybe for loop instead of forEach, because when you find a certain chat and set chat in setSelectedChat(chat) you can break iteration. forEach doesn't work break.
> 
> For example: It will be the first chat but you have fifty chats and when you use forEach it will run through the entire array.

Oh, you are absolutely right. Thank you for your advice! I think, using `listOfChats.find()` would be better than `listOfChats.for()`, because, `listOfChats.find()` performance wise is `O(n)`, and it would look much better in my opinion.